context/app/static/js/hooks/useSortState.ts

```diff
@@ -35,7 +33,7 @@ function sortReducer(state: SortState, action: SortAction): SortState {
         direction: 'desc',
       } as SortState;
     case 'reset':
-      return initialSortState;
+      return action.payload as SortState;
```

Comment:
Adding `payload: SortState` to the `type: "reset"` member of the `SortAction` union type should let you avoid the `as SortState`